File: ServerSentEvents/resources/reconnect-fail.php
<?php
  $id = "recon_fail_" . $_GET["id"];

  switch($_COOKIE[$id]) {
    case 'opened':
      header("HTTP/1.1 200 RECONNECT");
      header("Content-Type: text/event-stream");
      setcookie($id, "reconnected");
      echo "data: reconnected\n\n";
      break;

    case 'reconnected':
      header("HTTP/1.1 204 NO CONTENT (CLOSE)");
      header("Content-Type: text/event-stream");
      setcookie($id, ""); // Delete cookie
      echo "data: closed\n\n"; // Will never get through
      break;

    default:
      header("HTTP/1.1 200 OPEN");
      header("Content-Type: text/event-stream");
      setcookie($id, "opened");
      echo "retry: 2\n";
      echo "data: opened\n\n";
  }

?>

File: eventsource/resources/reconnect-fail.py
def main(request, response):
    name = b"recon_fail_" + request.GET.first(b"id")

    headers = [(b"Content-Type", b"text/event-stream")]
    cookie = request.cookies.first(name, None)
    state = cookie.value if cookie is not None else None

    if state == b'opened':
        status = (200, b"RECONNECT")
        response.set_cookie(name, b"reconnected");
        body = u"data: reconnected\n\n";

    elif state == b'reconnected':
        status = (204, b"NO CONTENT (CLOSE)")
        response.delete_cookie(name);
        body = u"data: closed\n\n" # Will never get through

    else:
        status = (200, b"OPEN");
        response.set_cookie(name, b"opened");
        body = u"retry: 2\ndata: opened\n\n";

    return status, headers, body

